docs/DeveloperGuide.md

```diff
@@ -159,6 +159,46 @@ After the `edit` command is called, the user input will be sent to **EditCommand
 #### Why it is implemented this way
 The `edit` command is implemented this way to ensure consistency with the other commands in the application. This helps to minimise any potential confusion for the users by standardising the prefixes that `edit` command takes in with the other relevant commands. In addition, it was intended for **EditCommandParser** to throw out a **ParseException** when none of the field changes an existing entry. This is to remind users in case they made a minor mistake, resulting in a supplier or warehouse to not update the way they intended for it to. Lastly, a command type prefix, `ct/COMMAND_TYPE` is required in the implementation of `edit` command to indicate whether user wishes to edit a warehouse or supplier entry. Without a command type prefix, an alternative would be for a `TYPE` parameter, where user have to indicate `supplier` or `warehouse`. However, this may not be suitable for our target user, who wishes to update stocks quickly. Hence, our team decided to use a command type prefix in place of a parameter. Furthermore, another alternative considered would be to create separate commands for warehouses and suppliers respectively. For example, `editw` and `edits` to represent edit warehouse and edit supplier. However, this might increase duplicated codes, since minimal changes to the code would be found for each class of command. By using a prefix, this helps us to reduce any potential code duplication. Therefore, our team decided to implement edit command by taking in prefixes and throwing our relevant exceptions at appropriate points after considering code quality and end user experience.
 
+### Update product feature
+
+The update product mechanism is facilitated by the `UpdateCommandParser` and the `UpdateCommand`.
+The `UpdateCommandParser` implements `Parser` and the `UpdateCommand` extends `Command`, allowing the user to 
+create or update a product under either a supplier or a warehouse.
+
+Given below is an example usage scenario and how the update product mechanism behaves at each step.
+
+Step 1. The user decides to update the stock for a product called 'Panadol' with a new quantity of 50 units 
+in the warehouse named 'Jurong Warehouse' by executing the `update ct/w n/Jurong Warehouse pd/Panadol q/50` command.
+The `ClinicParser#parseCommand` will then call the `UpdateCommandParser#parse` method with all the arguments 
+passed by the user.
+ 
+Step 2. `UpdateCommandParser#parse` then attempts to create new `Name` instances for the supplier/warehouse 
+and the product, and new `tag` instances for the product. An exception will be thrown if any of the names or 
```

Comment:
Hmm the example above doesn't really show the t/TAG! Might be a little confusing!

Reply:
Thats true, Ill update the example accordingly. Thanks!